okhttp/src/test/java/okhttp3/URLConnectionTest.kt

```diff
@@ -2937,12 +2942,14 @@ class URLConnectionTest {
     // required. These socket factories explicitly set the buffer sizes on sockets created.
     val socketBufferSize = 4 * 1024
     server.serverSocketFactory =
-      object : DelegatingServerSocketFactory(getDefault()) {
-        override fun configureServerSocket(serverSocket: ServerSocket): ServerSocket {
-          serverSocket.receiveBufferSize = socketBufferSize
-          return serverSocket
-        }
-      }
+      RealOkioServerSocketFactory(
+        object : DelegatingServerSocketFactory(getDefault()) {
```

Comment:
possibly a TODO? It would be nice to not need `DelegatingServerSocketFactory` and friends because of your beautiful new API.

Reply:
I'm not suggesting interceptors :) More that decorating could be made simpler?